controllers/ArticleController.js

```diff
@@ -0,0 +1,174 @@
+              body: [
+                'Ooops! the article cannot be found.'
+              ]
+            }
+          });
+        }
+
+        return res.status(200).json({ article });
+      })
+      .catch(() => res.status(501).send('oops  seems there is an error find the article'));
+  }
+
+  /**
+   * get all articles created
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} - the found article from database or empty if not found
+   */
+  static listAllArticles(req, res) {
+    return Article
+      .findAll({
+        include: [{
+          model: User,
+          attributes: { exclude: ['id', 'email', 'hashedPassword', 'createdAt', 'updatedAt'] }
+        }],
+        attributes: { exclude: ['id', 'userId'] }
+      })
+      .then((articles) => {
+        /** check if there was no article created */
+        if (articles.length === 0) {
+          return res.status(200).json({
+            message: 'Your request was successful but no articles created',
+            articles,
+          });
+        }
+
+        return res.status(200).json({ articles, articlesCount: articles.length });
+      })
+      .catch(() => res.status(501).send('oops  seems there is an error find all articles'));
+  }
+
+  /**
+   * @function editArticle
+  * @summary: API controller to handle requests
+  * to edit an article
+  * @param {object} req: request object
+  * @param {object} res: response object
+  * @returns {object} api response: article object for
+  * successful requests, or error object for
+  * requests that fail
+  */
+  static editArticle(req, res) {
+    const { title, description, body } = req.body.article;
+    const { count } = req;
+    const { slug } = req.params;
+    return Article.update({
+      title,
+      description,
+      body,
+      updatedCount: Utilities.increaseCount(count)
+    }, {
+      where: {
+        slug,
+      },
+      returning: true,
+      plain: true
+    })
+      .then(result => res.status(200).json({
+        success: true,
+        article: result[1]
+      }))
+      .catch((err) => {
+        res.status(500).json({
+          errors: {
+            body: [
+              'sorry there was an error updating this request',
+              err
+            ]
+          }
+        });
+      });
+  }
+
+  /**
+ * @function deleteArticle
+* @summary: API controller to handle requests
+* to delete an article
+* @param {object} req: request object
+* @param {object} res: response object
+* @returns {object} api response: article object for
+* successful requests, or error object for
+* requests that fail
+*/
+  static deleteArticle(req, res) {
+    const { slug } = req.params;
+    Article.destroy({
+      where: { slug }
+    })
+      .then(() => res.status(204).json())
+      .catch(() => {
+        res.status(500).json({
+          errors: {
+            body: [
+              'sorry there was an error updating this request',
```

Comment:
I think it should be "sorry there was an error deleting the Article"

Reply:
thank you for spotting that out out